docs/src/main/sphinx/sql/match-recognize.rst

```diff
@@ -0,0 +1,575 @@
+part of the match (plus the current row) is "visible". This is the ``running`` semantics.
+
+When evaluating expressions in ``MEASURES`` clause, the match is complete. It is then possible to apply the ``final``
+semantics. In the ``final`` semantics, the whole match is "visible" as from the position of the final row.
+
+In ``MEASURES`` clause, the ``running`` semantics can also be applied. When outputting information row by row
+(as in ``ALL ROWS PER MATCH``), the ``running`` semantics will evaluate expressions from the positions of consecutive
+rows.
+
+The ``running`` and ``final`` semantics are denoted by the keywords: ``RUNNING`` and ``FINAL``, preceding a logical navigation
+function ``first`` or ``last``::
+
+    RUNNING LAST(A.totalprice)
+
+    FINAL LAST(A.totalprice)
+
+The ``running`` semantics is default both in ``MEASURES`` and ``DEFINE`` clauses. ``FINAL`` can be only specified
+in ``MEASURES`` clause.
+
+With the option ``ONE ROW PER MATCH``, row pattern measures are evaluated from the position of the final row in the match.
+Therefore, ``running`` and ``final`` semantics are the same.
+
+aggregate functions
+^^^^^^^^^^^^^^^^^^^
+
+Standard SQL syntax allows to use aggregate functions inside pattern recognition expressions.
+Trino does not support it yet.
+
+Evaluating expressions in empty matches and unmatched rows
+----------------------------------------------------------
+
+An empty match occurs when the row pattern was successfully matched, but no pattern variables were assigned.
+The following pattern produces an empty match for every row::
+
+    PATTERN(())
+
+When evaluating row pattern measures for an empty match:
+
+- all column references return ``null``
+
+- all navigation operations return ``null``
+
+- ``classifier`` function returns ``null``
+
+- ``match_number`` function returns the sequential number of the match
+
+Like every match, an empty match has its starting row. All input values which are to be output along with the measures (as explained
+in :ref:`rows_per_match`), are the values from the starting row.
+
+An unmatched row is a row that is neither part of any non-empty match nor the starting row of an empty match.
+With the option ``ALL ROWS PER MATCH WITH UNMATCHED ROWS``, a single output row is produced. In that row, all row pattern measures
+are ``null``. All input values which are to be output along with the measures (as explained in :ref:`rows_per_match`), are the values
+from the unmatched row. Using the ``match_number`` function as a measure can help differentiate between an empty match and
+unmatched row.
+
+Future development
+------------------
+To support ``MATCH_RECOGNIZE`` syntax in full compatibility with the SQL standard, aggregate functions need to be supported in row pattern
+recognition expressions.
+
+Beside the ``MATCH_RECOGNIZE`` clause, SQL standard also permits pattern matching as part of :ref:`WINDOW clause<window_clause>`. The syntax
```

Comment:
again .. dont talk about future stuff. Just say what doesnt work..

Reply:
OK, so I'll only mention aggregations. Pattern matching in WINDOW cannot be considered a limitation, as it is an independent feature.